scalyr_agent/monitor_utils/k8s.py

```diff
@@ -17,6 +22,624 @@ class K8sApiException( Exception ):
     """
     pass
 
+class PodInfo( object ):
+    """
+        A collection class that stores label and other information about a kubernetes pod
+    """
+    def __init__( self, name='', namespace='', uid='', node_name='', labels={}, container_names=[], annotations={}, deployment_name=None ):
+
+        self.name = name
+        self.namespace = namespace
+        self.uid = uid
+        self.node_name = node_name
+        self.labels = labels
+        self.container_names = container_names
+        self.annotations = annotations
+        self.deployment_name = deployment_name
+
+        # generate a hash we can use to compare whether or not
+        # any of the pod info has changed
+        md5 = hashlib.md5()
+        md5.update( name )
+        md5.update( namespace )
+        md5.update( uid )
+        md5.update( node_name )
+
+        # flatten the labels dict in to a single string
+        flattened = []
+        for k,v in labels.iteritems():
```

Comment:
Is `iteritems` guaranteed to return the items in a deterministic order?  that impacts the hash.

Reply:
See previous comment